plugins/module_utils/common.py

```diff
@@ -318,7 +355,9 @@ def _wait_for_elapsed():
         response = None
         while _wait_for_elapsed() < timeout:
             try:
-                response = resource.get(name=name, namespace=namespace)
+                response = resource.get(name=name, namespace=namespace,
```

Comment:
One thing to note, I think previously the implementation of `wait` dealt only with a single resource, but multiple resources can be retrieved via a selector. We either need to iterate over the returned resources here or only pass individual resource names to the `wait` function.

Reply:
We could also implement a decorator for the predicates that deals with iterating over `ResourceList` objects if they're passed in. Any way works as long as we just iterate over the individual resources at some point.

Reply:
We should also add a test for the `k8s_info` wait that handles multiple objects matching the selector (ie, two pods that share a label)